src/elements/components/FeatheryAuthGate.tsx

```diff
@@ -0,0 +1,190 @@
+      history.replaceState(null, '', whitelistPath);
+    }
+
+    authState.clearLoader = () => setShowLoader(false);
+    // Register onLogin cb so it can be called by Client.submitAuthInfo
+    authState.onLogin = onLogin;
+    authState.onLogout = onLogout;
+
+    // If user passes authId as a prop we need to submit it
+    if (authIdProp) {
+      const defaultClient = new Client();
+      defaultClient.submitAuthInfo({
+        authId: authIdProp
+      });
+    }
+  }, []);
+
+  // This hook detects the login process then sets the redirect flag
+  // & auth loader
+  useEffect(() => {
+    if (
+      // We should set loader for new auth sessions
+      window.location.search.includes('stytch_token_type') ||
+      isHrefFirebaseMagicLink() ||
+      // and existing ones
+      getStytchJwt()
+    ) {
+      authState.redirectAfterLogin = true;
+      setShowLoader(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isAuthStytch()) return;
+
+    const authClient = getAuthClient();
+
+    const unsubscribe = authClient.session.onChange((newSession: any) => {
+      if (hasAuthedRef.current) {
+        if (newSession === null) logoutActions();
+        return;
+      }
+      // This onChange is just meant to set the token on initial login. It is
+      // not to cache the token. The token is retrieved directly every time it
+      // is needed by the function getStytchJwt
+      const newToken =
+        newSession?.stytch_session?.session_jwt ?? getStytchJwt();
+      if (!newToken) return;
+      hasAuthedRef.current = true;
+    });
+
+    return unsubscribe;
+  }, [render]);
+
+  const onActive = useCallback(
+    throttle(
+      () => {
+        const stytchClient = getAuthClient();
+        if (!stytchClient) return;
+
+        const session = stytchClient.session.getSync();
+
+        if (session) {
+          stytchClient.session.authenticate({
+            session_duration_minutes: 1440
+          });
+        } else if (hasAuthedRef.current) {
+          stytchClient.session.revoke().then(logoutActions);
+        }
+      },
+      FIVE_MINUTES_IN_MILLISECONDS,
+      { leading: true, trailing: false }
+    ),
+    []
+  );
+
+  useIdleTimer({
+    // Need to use lodash throttle rather than IdleTimer throttle because the
+    // IdleTimer throttle resets after the idle timeout. So, after 10 seconds of
+    // no activity the user moves the mouse and onActive fires as expected. Then
+    // if the idle timeout passes again and the user moves the mouse again,
+    // onActive will fire a second time, only 10 seconds after the first time,
+    // despite the 5 minute throttle. We need a low timeout to quickly detect
+    // if the user session has expired upon returning to feathery.
+    onActive,
+    timeout: TEN_SECONDS_IN_MILLISECONDS
+  });
+
+  const form = (
+    <>
+      {showLoader && (
+        <div
+          style={{
+            // TODO: what should the background color be?
```

Comment:
do you think we should provide a component prop for the loader background color? or just leave it as white until there is a demand for customizing the loader completely?

Reply:
will delete commented line below. left so you can see the old value since the code was moved

Reply:
i think it's fine to omit the component prop until it's needed. If this is the same loader component code as in the form, could we unify into a separate loader component? That maybe takes in a show boolean prop